modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/pagination/provider/PaginationProvider.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.vulcan.internal.pagination.provider;
+
+import com.liferay.portal.configuration.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.dao.orm.QueryUtil;
+import com.liferay.portal.kernel.module.configuration.ConfigurationException;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.vulcan.internal.configuration.HeadlessAPICompanyConfiguration;
+import com.liferay.portal.vulcan.pagination.Pagination;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Sergio Jiménez del Coso
+ */
+@Component(service = PaginationProvider.class)
+public class PaginationProvider {
+
+	public Pagination getPagination(
+		HttpServletRequest httpServletRequest, String page, String pageSize) {
```

Comment:
I think it should be better just to pass the companyId. I don't see the point of having the whole httpServletRequest here, as I don't see the point of having an HttpServletRequest to calculate the Pagination

Reply:
The point is that the `companyId` could be calculated using the `_portal` instance passing the `httpServletRequest`. Should the classes that implement this PaginationProvider know about the `_portal` instance passing by argument while we can calculate inject here? That's the reason to use the whole `httpServletRequest`. What do you think?